src/components/textbox/textbox.test.tsx

```diff
@@ -0,0 +1,607 @@
+
+    expect(screen.getByText("baz")).toHaveAttribute(
+      "id",
+      `${mockedGuid}-field-help`
+    );
+    expect(screen.getByRole("textbox")).toHaveAttribute(
+      "aria-describedby",
+      `${mockedGuid}-field-help`
+    );
+  });
+
+  test.each(["info", "warning", "error"])(
+    'when id is present, %s prop is set as a string, fieldHelp is present and the input is focused, the ids of both the validation tooltip are added to "aria-describedby" in the input',
+    (validationType) => {
+      render(
+        <Textbox
+          label="bar"
+          id="foo"
+          fieldHelp="baz"
+          {...{ [validationType]: "test" }}
+        />
+      );
+      const input = screen.getByRole("textbox");
+      input.focus();
+
+      expect(screen.getByRole("tooltip")).toHaveAttribute(
+        "id",
+        "foo-validation"
+      );
+      expect(screen.getByText("baz")).toHaveAttribute("id", "foo-field-help");
+      expect(input).toHaveAttribute(
+        "aria-describedby",
+        "foo-field-help foo-validation"
+      );
+    }
+  );
+
+  test.each(["info", "warning", "error"])(
+    'when id is not present, %s prop is set as a string, fieldHelp is present and the input is focused, the ids of both the validation tooltip are added to "aria-describedby" in the input',
+    (validationType) => {
+      render(
+        <Textbox
+          label="bar"
+          fieldHelp="baz"
+          {...{ [validationType]: "test" }}
+        />
+      );
+      const input = screen.getByRole("textbox");
+      input.focus();
+
+      expect(screen.getByRole("tooltip")).toHaveAttribute(
+        "id",
+        `${mockedGuid}-validation`
+      );
+      expect(screen.getByText("baz")).toHaveAttribute(
+        "id",
+        `${mockedGuid}-field-help`
+      );
+      expect(input).toHaveAttribute(
+        "aria-describedby",
+        `${mockedGuid}-field-help ${mockedGuid}-validation`
+      );
+    }
+  );
+});
+
+describe("when inputHint prop is present", () => {
+  test("renders the hint", () => {
+    render(<Textbox value="test string" inputHint="foo" />);
+    expect(screen.getByText("foo")).toBeInTheDocument();
+  });
+
+  test("assigns the input hint a guid as id and references it in the aria-describedby of the input", () => {
+    render(<Textbox value="test string" inputHint="bar" />);
+    expect(screen.getByText("bar")).toHaveAttribute("id", mockedGuid);
+    expect(screen.getByRole("textbox")).toHaveAttribute(
+      "aria-describedby",
+      mockedGuid
+    );
+  });
+
+  test("inputHint should have priority over labelHelp when both are passed", () => {
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <Textbox labelHelp="labelHelp" inputHint="inputHint" error="foo" />
+      </CarbonProvider>
+    );
+    expect(screen.getByText("inputHint")).toBeInTheDocument();
+    expect(screen.queryByText("labelHelp")).not.toBeInTheDocument();
+  });
+});
+
+describe("new validations", () => {
```

Comment:
suggestion(non-blocking): this describe block isn't adding much value

We should probably either be more verbose with the description or drop the describe block